lib/worker/src/routes/vaults.rs

```diff
@@ -0,0 +1,460 @@
+
+    HttpResponse::Ok().json(CreateVaultResponse { created })
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateVaultInput {
+    pub cache: Option<i64>,
+    pub account: Option<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct CreateVaultResponse {
+    created: bool,
+}
+
+pub async fn write_record(
+    path: web::Path<String>,
+    params: web::Query<WriteRecordParams>,
+    mut payload: web::Payload,
+    gcs_client: web::Data<GcsClient>,
+    pool: web::Data<PgPool>,
+) -> HttpResponse {
+    let vault: Vault = match path.try_into() {
+        Ok(p) => p,
+        Err(err) => {
+            return HttpResponse::BadRequest().json(ErrorResponse {
+                error: err.to_string(),
+            })
+        }
+    };
+
+    match db::namespace_exists(&pool, &vault.namespace()).await {
+        Ok(exists) => {
+            if !exists {
+                return HttpResponse::NotFound().json(ErrorResponse {
+                    error: "namespace not found".to_string(),
+                });
+            }
+        }
+        Err(err) => {
+            println!("{}", err);
+            log::error!("{}", err);
+            return HttpResponse::InternalServerError().json(ErrorResponse {
+                error: "error checking the vault".to_string(),
+            });
+        }
+    }
+
+    let cache_duration =
+        match db::get_cache_config(&pool, &vault.namespace(), &vault.relation()).await {
+            Ok(v) => v,
+            Err(err) => {
+                log::error!("{}", err);
+                return HttpResponse::InternalServerError().json(ErrorResponse {
+                    error: "error fetching the cache config".to_string(),
+                });
+            }
+        };
+
+    let mut metadata: HashMap<String, String> = HashMap::from([(
+        "timestamp".into(),
+        format!(
+            "{}",
+            params
+                .timestamp
+                .unwrap_or(chrono::offset::Utc::now().timestamp())
+        ),
+    )]);
+
+    if cache_duration.is_some() {
+        metadata.insert(
+            "cache_duration".into(),
+            format!("{}", cache_duration.unwrap()),
+        );
+    }
+
+    let filename = format!(
+        "{}/{}/{}.parquet",
+        vault.namespace(),
+        vault.relation(),
+        chrono::Utc::now().timestamp_micros()
+    );
+    let upload_type = UploadType::Multipart(Box::new(Object {
+        name: filename.to_string(),
+        content_type: Some("application/octet-stream".into()),
+        metadata: Some(metadata),
+        ..Default::default()
+    }));
+
+    let uploader = gcs_client
+        .inner
+        .prepare_resumable_upload(
+            &UploadObjectRequest {
+                bucket: gcs_client.bucket.clone(),
+                ..Default::default()
+            },
+            &upload_type,
+        )
+        .await
+        .unwrap();
+
+    let mut hasher = Keccak::v256();
+
+    while let Ok(Some(bytes)) = payload.try_next().await {
```

Comment:
Is the payload size fixed? Can we keep it constant for the hashing purpose? For instance always reading  1MB chunks and hashing those one by one. That will be quite helpful when we move to ECMH